### ISSUE TITLE
FaceShifter: Towards High Fidelity And Occlusion Aware Face Swapping

### ISSUE THREAD

**Issue by Yagami360** [Yagami360/MachineLearning-Papers_Survey#106]
## 0. 論文情報・リンク
- 論文リンク：https://arxiv.org/abs/1912.13457
- 公開日時：2019/12/31
- 組織 : Microsoft Research
- 被引用数（記事作成時点）：8 件
- 実装コード（PyTorch）：
    - https://github.com/taotaonice/FaceShifter
    - https://github.com/mindslab-ai/faceshifter
    - https://github.com/Heonozis/FaceShifter-pytorch
プロジェクトページ : https://lingzhili.com/FaceShifterPage/
- Publication : CVPR 2020

## 1. どんなもの？
- AEI-Net と HEAR-Net の２段階のアーキテクチャにより、オクリュージョンがある場合にも変換元と変換先の２つの画像の属性の同一性を不自然なく合成することを実現した GAN ベースの顔すり替え（Face Swapping）モデル
    <image src="https://user-images.githubusercontent.com/25688193/97300771-3e0cf980-189a-11eb-8d4f-589099f736b8.png" width="970">

## 2. 先行研究と比べてどこがすごいの？
- 顔すり替えタスクでは、変換元と変換先の２つの画像の人物同一性や属性（ポーズ・表情・背景など）を不自然なく合成することが大きな課題となっている。
	しかしながら既存の顔すり替えモデルでは、GAN を用いたアプローチでも、上記課題を克服することができていない。
	本手法は、まず AEI-Net で変換元と変換先の２つの画像の同一性を不自然なく合成し、次に HEAR-Net でオクリュージョンがある場合にも対処する２段階のアーキテクチャで、オクリュージョンがある場合にも変換元と変換先の２つの画像の人物同一性や属性を不自然なく合成することを実現した GAN ベースの顔すり替えモデルになっている。

## 3. 技術や手法の"キモ"はどこにある？
- アーキテクチャの全体像
    <image src="https://user-images.githubusercontent.com/25688193/97300886-5da42200-189a-11eb-8f35-13ecc4cdd29d.png" width="800">

	上図は、FaceShifter のアーキテクチャの全体像である。
	FaceShifterは、以下の２つの主要コンポーネントから構成される。
	
	1. AEI-Net [Adaptive Embedding Integration Network]
		｛目標顔画像（すり替えベース顔）・ソース顔画像（すり替え顔）｝のペアデータから、ソース顔画像の人物同一性を維持したまま、目標顔画像の属性（ポーズ・表情・背景など）を取り込んだオクリュージョン対処されていないすり替え顔画像を生成するネットワーク。
		
	2. HEAR-Net [Heuristic Error Acknowledging Network]
目標顔画像とソース顔画像を同じにして AEI-Net で生成した生成画像とその正解画像との差分マップ [heuristic error] を利用することで、オクリュージョン対処されたすり替え顔画像を生成するネットワーク。

- AEI-Net [Adaptive Embedding Integration Network]
    <image src="https://user-images.githubusercontent.com/25688193/97301335-ff2b7380-189a-11eb-9a17-a918a0de7f3c.png" width="850">

	AEI-Net では、｛目標顔画像（すり替えベース顔）・ソース顔画像（すり替え顔）｝のペアデータから、ソース顔画像の人物同一性を維持したまま、目標顔画像の属性（ポーズ・表情・背景など）を取り込んだオクリュージョン対処されていないすり替え顔画像を生成する。
	
	AEI-Net は、以下の３つのサブコンポーネントから構成される。
	1. Identity Encoder : <image src="https://user-images.githubusercontent.com/25688193/97301456-23875000-189b-11eb-8bdc-f650147b81f6.png" width="47">
		ソース顔画像 X_s から人物同一性（Identity）を抽出するエンコーダー
	2. Multi-level Attributes Encoder : <image src="https://user-images.githubusercontent.com/25688193/97301530-3bf76a80-189b-11eb-904c-d975da8a0504.png" width="57">
		目標顔画像 X_t から属性（ポーズ・表情・背景など）を抽出するエンコーダー
	3. AAD [Adaptive Attentional De-normalization] Generator
		Identity Encoder と Multi-level Attributes Encoder で抽出した特徴量から、顔すり替え画像を生成する生成器

    - Identity Encoder
        <image src="https://user-images.githubusercontent.com/25688193/97301655-5f221a00-189b-11eb-8abd-175e55e5e978.png" width="211">
		エンコーダーネットワークとしては、顔認識モデルの SOTA である事前学習された Arcface ネットワークを利用する。
		但し Identity Encoder で出力する特徴マップは、Arcface ネットワークの最後の FC 層の前段から出力される特徴マップを採用する。
		※ 論文「Arcface: Additive angular margin loss for deep face recognition」

    - Multi-level Attributes Encoder
        <image src="https://user-images.githubusercontent.com/25688193/97301759-84af2380-189b-11eb-8244-3c60380d3084.png" width="400">
		画像から属性（ポーズ・表情・背景など）を抽出するためには、人物同一性のときよりも多くの（画像中の）空間情報を抽出する必要がある。
		そのため Identity Encoder のように単一の特徴マップとして特徴抽出するのではなく、複数の層からの特徴マップとして特徴抽出を行う。
		具体的には、下式のように UNet の各デコーダー層からの特徴マップで特徴抽出を行う。
        <image src="https://user-images.githubusercontent.com/25688193/97301836-95f83000-189b-11eb-8a19-7478c6613fb1.png" width="375">

    - AAD [Adaptive Attentional De-normalization] Generator
        <image src="https://user-images.githubusercontent.com/25688193/97302163-07d07980-189c-11eb-904e-8187a6961aa4.png" width="460">
		AAD では、Identity Encoder と Multi-level Attributes Encoder で抽出した特徴量から、顔すり替え画像を生成する。
		このとき、ネットワークの入力部のみにこれら特徴量を concat して入力する方法では、生成される顔すり替え画像がぼやけてしまうので、SPADE や AdaIN のようにこれら抽出した特徴量をネットワークの各段に入力する。
		
		ここで AAD 内の処理は、以下のようになる。

		① BatchNorm 
			<image src="https://user-images.githubusercontent.com/25688193/97302254-259dde80-189c-11eb-915f-51b8ff15afef.png" width="229">
			まず k 番目の AAD ResBlock 層の ADD への活性化入力 hin^k  に対して、以下の式で定義される BatchNorm を行う。
			<image src="https://user-images.githubusercontent.com/25688193/97302301-36e6eb00-189c-11eb-88f7-0f93083a33c2.png" width="263">

		② attributes integration 
			次に、BatchNorm  した活性化入力 h̅^k  に対して、AdaIN や SPADE の正則化層のように、以下のアフィン変換の式で定義される正則化処理を行う。
			<image src="" width="">
			<image src="" width="">

    <image src="" width="">

## 4. どうやって有効だと検証した？
- 学習用データセット
- 実験設定
- 定性検証
- 定量検証
    <image src="" width="">

## 5. 議論はあるか？
- xxx

## 6. 次に読むべき論文はあるか？
- xxx

## 7. 参考文献
- xxx
